src/main/kotlin/no/nav/familie/ef/sak/økonomi/TilkjentYtelseService.kt

```diff
@@ -0,0 +1,159 @@
+package no.nav.familie.ef.sak.økonomi
+
+import no.nav.familie.ef.sak.økonomi.dto.*
+import no.nav.familie.kontrakter.felles.Ressurs
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsoppdrag
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+
+@Service
+class TilkjentYtelseService(
+        private val økonomiKlient: ØkonomiKlient,
+        private val tilkjentYtelseRepository: TilkjentYtelseRepository,
+        private val andelTilkjentYtelseRepository: AndelTilkjentYtelseRepository
+) {
+    @Transactional
+    fun opprettTilkjentYtelse(tilkjentYtelseRestDTO : TilkjentYtelseRestDTO) : Long {
+        tilkjentYtelseRestDTO.valider()
+
+        val tilkjentYtelse = tilkjentYtelseRestDTO.tilTilkjentYtelse(TilkjentYtelseStatus.OPPRETTET)
+        val tilkjentYtelseId = tilkjentYtelseRepository.save(tilkjentYtelse).id
+
+        andelTilkjentYtelseRepository.saveAll(tilkjentYtelseRestDTO.tilAndelerTilkjentYtelse(tilkjentYtelseId))
+
+        return tilkjentYtelseId
+    }
+
+    @Transactional
+    fun iverksettUtbetalingsoppdrag(tilkjentYtelseId: Long, saksbehandlerId: String) {
+        val tilkjentYtelse = hentTilkjentYtelse(tilkjentYtelseId)
+
+        when(tilkjentYtelse.status) {
+            TilkjentYtelseStatus.SENDT_TIL_IVERKSETTING -> return
+            TilkjentYtelseStatus.AKTIV -> return
+            TilkjentYtelseStatus.IKKE_KLAR -> throw IllegalStateException("Tilkjent ytelse $tilkjentYtelseId er ikke klar")
+            TilkjentYtelseStatus.AVSLUTTET -> throw IllegalStateException("Tilkjent ytelse $tilkjentYtelseId er avsluttet")
+            TilkjentYtelseStatus.OPPRETTET -> {
+                val andelerTilkjentYtelse = hentAndelerTilkjentYtelse(tilkjentYtelseId)
+
+                val utbetalingsoppdrag = lagUtbetalingsoppdrag(saksbehandlerId, tilkjentYtelse, andelerTilkjentYtelse)
+                iverksettOppdrag(utbetalingsoppdrag)
+
+                // Mulighet for inkonsistens hvis exception her, eller neste steg feiler
+                tilkjentYtelseRepository.save(tilkjentYtelse.copy(
+                        utbetalingsoppdrag = objectMapper.writeValueAsString(utbetalingsoppdrag),
+                        status = TilkjentYtelseStatus.SENDT_TIL_IVERKSETTING
+                ))
```

Comment:
Hvis vi løser dette motsatt vei kan vi iallefall rulle tilbake:
- Oppdater intern tilstand først
- Send `iverksettOppdrag` etterpå

Hvis første feiler, så sender vi aldri noe oppdrag.
Hvis siste feiler klarer vi å rulle tilbake. 

Ulempen med dette er jo at det "logisk sett" feil rekkefølge. 

Kanskje andre har idéer

Reply:
Tror jeg begynner med @charliemidtlyng sitt forslag. Et alternativ er å benytte prosess-motoren, for å få overvåkning, retries etc